### PR TITLE
refactor: Move DeltaManagerProxyBase from container-utils to container-runtime

### PR DESCRIPTION
**Pull request by Josmithr**
We are working to remove the `@fluidframework/container-utils` package, as we do not intend / expect external consumers to use it.

As a part of this effort, we have identified that the `DeltaManagerProxyBase` class is only used in a single place in our repo, and not used externally (based on a search of Bohemia, WordJs, PowerPoint Online, and Whiteboard). This PR moves the class into `@fluidframework/container-runtime`, which is the only place that currently uses it. It is no longer exported.

### BREAKING CHANGE (container-utils)

This PR removes the `DeltaManagerProxyBase` from the API. There is no intended replacement.

### CODE REVIEW

**Review by ChumpChief — APPROVED** [review]
Nice, why didn't I think of that? :-D

File: .changeset/sour-jobs-count.md
Comment (tylerbutler) [inline .changeset/sour-jobs-count.md:8]: Was this deprecated already? If not, can we deprecate it in 6.1?

Comment (Josmithr) [reply .changeset/sour-jobs-count.md:8]: Not yet, no.  None of our partners are using it, but I suppose it would still be good form to do that. I'll put up a separate PR doing that in main.

Comment (Josmithr) [reply .changeset/sour-jobs-count.md:8]: PR deprecating the API on main: #16813

Comment (Josmithr) [reply .changeset/sour-jobs-count.md:8]: I'll update the details in this PR once the other PR has gone through and been merged into next.

Comment (Josmithr) [reply .changeset/sour-jobs-count.md:8]: PR merged into main. Will wait until it has been merged into next before updating this PR.

Comment (Josmithr) [reply .changeset/sour-jobs-count.md:8]: I messed up a merge. I'm just going to open a fresh PR rather than trying to clean up the history.

Comment (Josmithr) [reply .changeset/sour-jobs-count.md:8]: Replacement PR: #16875

**Pull request closed without merging**
